vmm/src/config.rs

```diff
@@ -1197,6 +1210,57 @@ impl SgxEpcConfig {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize, Default)]
+pub struct NumaConfig {
+    #[serde(default)]
+    pub id: u64,
+    #[serde(default)]
+    pub cpus: Option<Vec<u64>>,
+    #[serde(default)]
+    pub destination: Option<Vec<u64>>,
+    #[serde(default)]
+    pub distance: Option<Vec<u64>>,
+}
+
+impl NumaConfig {
+    pub const SYNTAX: &'static str = "Settings related to a given NUMA node \
+        \"id=<node_id>,cpus=<cpus_id>,destination=<list_of_destination_nodes>,\
+        distance=<list_of_distances_to_destination_nodes>\"";
```

Comment:
Rather than having two lists that need to be the same length (destination and distance). Why not zip them together? e.g. <node1>:<distance1>,<node2>:<distance2>. Makes the validation easier.